Initial submission of annotation-model framework

halindrome opened the pull request:
This is a collection of scripts that the Web Annotation working group is using to author its tests.  There are a couple of example manual tests in the tree to show how the basic flow will work.  The actual tests are being developed separately right now but we would like to migrate into the main WPT repo as soon as possible.

This is my first attempt at adding to WPT, so I expect I did something wrong.  Let me know!

halindrome committed:
Initial creation.

halindrome committed:
Initial creation.

Added handling for skipping and/or aborting

Updated some documentation.

Refactored external assertion loading

Now loads all the assertion files ahead of time, then structures the
expanded assertions into an Array that is iterated over at run time.

This will allow the easy extraction of the assertion information so that
we can structure the test instructions to show what is expected.

Added line to permit console statements in ajv

Added a --examples option

If selected, then the tests in --examples are generated.  Otherwise they
are not.

Added a new examples directory

These are here for documentation in preparation for
releasing the framework to WPT for review.

Working version with assertion and condition objects.

also split documentation.  Ready to send as PR.

Cleaned up lint errors.

Enabled the v5 compatibilty flag in ajv

This will allow (among other things) the contains
keyword, which will help is avoid the not ... not nonsense.

Added automatic processing of definitions

As per working group decision from Friday, 24 June.

Also disabled schema v5 processing as per working group
decision from Friday, 24 June.

updated examples

Added handling for failure to load component

Now creates a test that is marked failed for any components
that fail to load at test startup.

halindrome committed:
Merge remote-tracking branch 'upstream/master'

halindrome committed:
Initial structure.

halindrome committed:
Merge branch 'master' of github.com:w3c/web-platform-tests

halindrome committed:
Merge branch 'shane_json1'

halindrome committed:
Started defining syntax.

Created an example manual test and started to populate it.
Pulled in a single "assertion" json schema file from @bigbluehat

Incorporated the most recent version of ajv so we are freestanding.

halindrome committed:
Added a config so stuff gets styled the same.

halindrome committed:
Initial version.

Functions to load and process JSON tests.

halindrome committed:
Changed to encapsulate JSON Schema in a test wrapper.

halindrome committed:
Renamed so that it doesn't get picked up in manifest

halindrome committed:
Updated documentation based upon current thinking.

Added multiple assertions per test file.
Allowed for validation schema to be embedded or external.

halindrome committed:
Fixed typos.

halindrome committed:
Initial versions of library and samples

Update of documentation.
First "common" JSON file.

halindrome committed:
Merge remote-tracking branch 'upstream/master'

halindrome committed:
First working example.

Code works - iterates over assertions and tests against input.
Needs a lot of testing.

halindrome committed:
Some updates to Assertion Object structure

halindrome committed:
Added handling for skipping and/or aborting

Updated some documentation.

halindrome committed:
Added some additional examples.

Also enhanced documentation and added some placeholder
common schema.

halindrome committed:
Re-wrapped so it looks better unformatted.

halindrome committed:
Added a simple model test

Also removed original samples

halindrome committed:
Fixed a typo.

halindrome committed:
Initial working version

Builds tests from .test files

halindrome committed:
Created new sample files

Simple example .test files

halindrome committed:
Fixed a typo.

halindrome committed:
Added error output if JSON parsing fails

halindrome committed:
Changed loading to use Promises

Now loads test and any dependent assertion files in parallel
while waiting for DOMContentReady.  Once everything is loaded, enables
the test and sets the run button to active.

halindrome committed:
Fixed issue with assertionCache.

Also fixed the test button text.

halindrome committed:
Refactored external assertion loading

Now loads all the assertion files ahead of time, then structures the
expanded assertions into an Array that is iterated over at run time.

This will allow the easy extraction of the assertion information so that
we can structure the test instructions to show what is expected.

halindrome committed:
Added a task

halindrome committed:
Added logic to emit Assertion tree to the test page.

halindrome committed:
Changed the template to match script expectations.

@@@TODO@@@ this needs documentation.

halindrome committed:
Added diagnostic output and title updating

tcole3 committed:
broke the id property check into 2 schemas

tcole3 committed:
create the second id assertion object.

tcole3 committed:
Rename idIsUri to idIsUri.json

tcole3 committed:
Create typeAnnotation.json

tcole3 committed:
must have @context, and its value must be correct

tcole3 committed:
Separate check for @context from value check

Separated check for whether Annotation has at least 1 @context value from check of whether @context values include link to anno.jsonld.

tcole3 committed:
Separately checks value of @context

Note - no error is reported if @context is missing altogether, since this is checked by a different JSON schema. This separation is more germane when the property value is a MUST but including the property is only a SHOULD.

tcole3 committed:
corrected name of schema file

Note - the spec requires id be an IRI, but at the moment the closest format match is URI.

tcole3 committed:
updated title and error message

tcole3 committed:
Updated to check that only 1 id

tcole3 committed:
corrected typo and updated title

tcole3 committed:
Separated check for type from check of value

type maps to rdf:type which has range rdfs:Class which in JSON we assume is string of format uri, i.e., type can only be a string or an array of strings.

tcole3 committed:
corrected error message

tcole3 committed:
made additionalItems explicit

tcole3 committed:
Check value of type

tcole3 committed:
check for at least a body or bodyText

tcole3 committed:
fails if both body and bodyText

Also fails if neither is present

tcole3 committed:
Copied correct file this time

Also fails if neither body nor bodyText is present.

tcole3 committed:
objects are allowed as targets

This schema now just checks that target property is present, it does not check that target property has a valid value. Left to schemas yet to be written.

tcole3 committed:
removed conflation of bodyText type check

tcole3 committed:
removed type check on bodyText

tcole3 committed:
removed type check on id

tcole3 committed:
fixed error message

tcole3 committed:
First try at defining test for Section 3.1

tcole3 committed:
first try at more granular test

tcole3 committed:
Changed should to failAndSkip

While not meeting a Should is not a failure in terms of validation, it is a failure in terms of implementing the feature, so decided to change the onUnexpectedResult value from passAndSkip to failAndSkip for body not present and id not present.

halindrome committed:
Corrected example.

halindrome committed:
Escaped periods in REs

halindrome committed:
Changed logic to enforce value check as per spec.

halindrome committed:
Don't disable the check button on run.

halindrome committed:
Added a note about "or" handling.

halindrome committed:
Added some tasks.

halindrome committed:
Merged in corrected context test

This test allows the context value to be anywhere in the list.

halindrome committed:
Enabled the v5 compatibilty flag in ajv

This will allow (among other things) the contains
keyword, which will help is avoid the not ... not nonsense.

halindrome committed:
Removed because they are hosted in another repo

halindrome committed:
Removed these - we don't need them now.

halindrome committed:
Interim version.

halindrome committed:
Merge remote-tracking branch 'upstream/master'

engelke committed:
Merge pull request #2 from w3c/master

Catch up to W3C master

halindrome committed:
Added line to permit console statements in ajv

engelke committed:
Merge pull request #3 from w3c/master

Catch up to W3C

halindrome committed:
Fixed a typo.

engelke committed:
Merge pull request #4 from w3c/master

Update

halindrome committed:
Added a --examples option

If selected, then the tests in --examples are generated.  Otherwise they
are not.

halindrome committed:
Added a new examples directory

These are here for documentation in preparation for
releasing the framework to WPT for review.

halindrome committed:
Working version with assertion and condition objects.

also split documentation.  Ready to send as PR.

halindrome committed:
Cleaned up lint errors.

halindrome committed:
Merge remote-tracking branch 'upstream/master'

halindrome committed:
Merge branch 'master' of github.com:Spec-Ops/web-platform-tests

halindrome committed:
Merge branch 'master' into add_or_handling

halindrome committed:
Merge pull request #4 from Spec-Ops/add_or_handling

Add or handling

Created examples folder

Removed folders that conflict with working group repo

halindrome committed:
Merge remote-tracking branch 'upstream/master'

engelke committed:
Merge pull request #5 from w3c/master

Catch up to W3C master

BigBlueHat commented on annotation-model/TODO:
```diff
@@ -0,0 +1,20 @@
+
+Add a Skip option to tools/runner
+
+Add a "or" mode for collections of assertions.
```
@halindrome I believe the Skip logic and possibly the "or" situation are taken care of now, correct?

halindrome replied:
The skip is something I want to add to runner - right now there is no way to say "I don't want to run this test right now."  

The or mode is supported in the comparisonObject handler.  I will update the TODO file.  Thanks!

BigBlueHat commented on browser-payment-api/OWNERS:
```diff
@@ -0,0 +1 @@
+@halindrome
```
@halindrome looks like this snuck in from a different branch? :smiley: :

halindrome replied:
Wow - weird!  Thanks

BigBlueHat commented:
Other than some very small nits, this looks great. Thanks, @halindrome!

:+1:

halindrome committed:
Fixed todo as per @bigbluehat

Also removed browser-payment-api stuff.  That is for a different
project!

halindrome committed:
Added automatic processing of definitions

As per working group decision from Friday, 24 June.

Also disabled schema v5 processing as per working group
decision from Friday, 24 June.

halindrome committed:
updated examples

halindrome committed:
Reset head.

halindrome committed:
Merge remote-tracking branch 'upstream/master'

halindrome committed:
Merge branch 'master' of github.com:Spec-Ops/web-platform-tests

halindrome commented:
There was a comment from @tcole3 in a Web Annotation working group meeting that all the .json files in the definitions directory be included into the ajv environment automatically.  This is now implemented via commit https://github.com/w3c/web-platform-tests/pull/3173/commits/e36c68f89a4f7683b20c66fa29d288ee87f49d1e

halindrome committed:
Added handling for failure to load component

Now creates a test that is marked failed for any components
that fail to load at test startup.

youennf committed:
XHR empty mime type blob test

alastor0325 committed:
fix web-platform-test 'webvtt-api-for-browsers/vttcue-interface/vertical.html'.

MozReview-Commit-ID: G0uKsIw5Mx5

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1278478

alastor0325 committed:
part3 : modify web-platform test.

MozReview-Commit-ID: BjeiP1XeC8T

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1276130

alastor0325 committed:
part4 : modify web-platform-test.

The idl-test-harness can't detect the multiple types in webidl [1], so it doesn't know about the type like "double or AutoKeyword".
The same problem also happen in VTTCue's line property, and we have already expected its fail.
Therefore, we should expect this fail before we fix the problem of idl-test-harness.

[1] http://searchfox.org/mozilla-central/source/dom/imptests/idlharness.js#375
MozReview-Commit-ID: 5XvOwdmqKDP

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1276831

upsuper committed:
Apply fullscreen enabled flag to about:blank as well.

MozReview-Commit-ID: JL3ZIdXpLiX

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1279613

bechen committed:
Fix web-platform/meta/html/semantics/embedded-content/media-elements/interfaces/TextTrack/cues.html.

MozReview-Commit-ID: 7VuUfyegq2d

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1278151

bzbarsky committed:
Ensure that when unblocking the scriptloader we try to process scripts if we have _any_ scripts to process, not just parser-blocking ones.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1281366

bzbarsky committed:
Fire error events for <object> per spec.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1003797

jgraham committed:
Import entities.json file to avoid external network fetch in wpt xml test,

MozReview-Commit-ID: IxniFEeNGxw

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1279810

youennf committed:
Update fetch cors tests to limit the use of www/www1 URLs

jgraham committed:
Split WebCryptoAPI generateKey tests into one test file per algorithm name

MozReview-Commit-ID: FVHQoZDV5eI

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1279799

jgraham committed:
Add a long timeout to some slow-running editing tests,

MozReview-Commit-ID: 5fg9dcnLbI7

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1279808

jgraham committed:
Add a content-type to form-submission script response,

MozReview-Commit-ID: 5HT7PnRy24l

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1279830

karlt committed:
test for bug 1213313 r=padenot

MozReview-Commit-ID: 6CU9AgW8dZx

karlt committed:
test for bug 1257718 r=padenot

MozReview-Commit-ID: 8MkfnpsQBKH

Commit:
Update and add test cases.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264165

mantaroh committed:
Add DocumentTimeline constructor tests.

MozReview-Commit-ID: A6vSuodpWGQ

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267510

mantaroh committed:
Tidy up to document-timeline tests.

MozReview-Commit-ID: DKlmsrKWF6K

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267510

wisniewskit committed:
Correct XHR web platform test to verify that the last-modified time is set correctly on the response document if it is set in the HTTP headers (or not).

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1280454

BorisChiou committed:
Test for creating animations with a specific spacing mode.

MozReview-Commit-ID: J0xNLunv9A7

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1244590

BorisChiou committed:
Test for different targets.

MozReview-Commit-ID: B0hrEKRdKEG

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1244590

jgraham committed:
Merge pull request #3244 from w3c/sync_619e43026f209f02b741adf074c638ad01b77c45

Merge pull request #3244 from sync_619e43026f209f02b741adf074c638ad01b77c45

jgraham committed:
Merge pull request #3245 from w3c/sync_6c186c61d79395b24b78528734211a35f00bfc89

Merge pull request #3245 from sync_6c186c61d79395b24b78528734211a35f00bfc89

jgraham committed:
Merge pull request #3246 from w3c/sync_5b0574f073bb8f74e3ddebfcf3da656ce4d724dd

Merge pull request #3246 from sync_5b0574f073bb8f74e3ddebfcf3da656ce4d724dd

jgraham committed:
Merge pull request #3247 from w3c/sync_7dfd2e80f5194ddf1912c6c2bca75b868f35413f

Merge pull request #3247 from sync_7dfd2e80f5194ddf1912c6c2bca75b868f35413f

jgraham committed:
Merge pull request #3248 from w3c/sync_273361c68fa6a3b69ce1565fd6ea9a6ce2a52067

Merge pull request #3248 from sync_273361c68fa6a3b69ce1565fd6ea9a6ce2a52067

jgraham committed:
Merge pull request #3249 from w3c/sync_a0a077096a5dea859321b7ce0fa1c8ce992d8bab

Merge pull request #3249 from sync_a0a077096a5dea859321b7ce0fa1c8ce992d8bab

jgraham committed:
Merge pull request #3250 from w3c/sync_41b6ebef64c9ce04678e97671801eac8cbb2c441

Merge pull request #3250 from sync_41b6ebef64c9ce04678e97671801eac8cbb2c441

jgraham committed:
Merge pull request #3251 from w3c/sync_7a5872cebb4be41801818692edf7bb952e1f0356

Merge pull request #3251 from sync_7a5872cebb4be41801818692edf7bb952e1f0356

jgraham committed:
Merge pull request #3252 from w3c/sync_4ab1a7932360568b06c982954e52b3696e962d6c

Merge pull request #3252 from sync_4ab1a7932360568b06c982954e52b3696e962d6c

jgraham committed:
Merge pull request #3253 from w3c/sync_658876eaec72b89d13370a933557ad3bd625e0ad

Merge pull request #3253 from sync_658876eaec72b89d13370a933557ad3bd625e0ad

jgraham committed:
Merge pull request #3254 from w3c/sync_2fb6781fb16b34c2fc8c0702c7a10e3038446916

Merge pull request #3254 from sync_2fb6781fb16b34c2fc8c0702c7a10e3038446916

jgraham committed:
Merge pull request #3255 from w3c/sync_e1a04e1ddd2e5d494b8179146a221abb7c1312b2

Merge pull request #3255 from sync_e1a04e1ddd2e5d494b8179146a221abb7c1312b2

jgraham committed:
Merge pull request #3256 from w3c/sync_1242f41cc9857ae57d32e58f4245ff664360ec5e

Merge pull request #3256 from sync_1242f41cc9857ae57d32e58f4245ff664360ec5e

jgraham committed:
Merge pull request #3257 from w3c/sync_5493519f9248a665b38c58cf32b5f07b3dcaa882

Merge pull request #3257 from sync_5493519f9248a665b38c58cf32b5f07b3dcaa882

jgraham committed:
Merge pull request #3258 from w3c/sync_761f3ad91b457829359c2b299db5d2c90c19e7ec

Merge pull request #3258 from sync_761f3ad91b457829359c2b299db5d2c90c19e7ec

jgraham committed:
Merge pull request #3259 from w3c/sync_b01834d68475e18937eed9218ced3bff6e2407db

Merge pull request #3259 from sync_b01834d68475e18937eed9218ced3bff6e2407db

jgraham committed:
Merge pull request #3260 from w3c/sync_fe9bcee6dd88a48d0341ee7afe3db41eafed67ad

Merge pull request #3260 from sync_fe9bcee6dd88a48d0341ee7afe3db41eafed67ad

jgraham committed:
Merge pull request #3261 from w3c/sync_fe8a446d1d27f961346581e252c577e245d45650

Merge pull request #3261 from sync_fe8a446d1d27f961346581e252c577e245d45650

jgraham committed:
Merge pull request #3262 from w3c/sync_8ef9f48e673770015af37c7e86d95719060e6ab9

Merge pull request #3262 from sync_8ef9f48e673770015af37c7e86d95719060e6ab9

jdm committed:
Merge pull request #3241 from youennf/fix-blob-empty-mime-type

XHR empty mime type blob test

RByers committed:
Update pointerevents README and add myself as an OWNER (#3263)

NavidZ committed:
Add pointer events test for pointerId across frames (#3131)

We need a test to make sure the pointerId of an active pointer is the same across iframes

NavidZ committed:
Adding more complete button & buttons attribute test (#3181)

louaybassbouss committed:
presentation-api idlharness update (#3238)

* * update idlharness.html with latest Presentation API interfaces from the ED

* * update idlharness for presentation API

youennf committed:
Removing https

jdm committed:
Merge pull request #3242 from youennf/update-cors-tests-to-limit-use-of-www-www1-urls

Update fetch cors tests to limit the use of www/www1 URLs

jdm committed:
Allow pages to finish loading when external stylesheets have network errors.

Commit:
Added checks that selection is valid, and fix set_selection_range.

jdm committed:
Ensure that a navigation to the same URL is aborted. Fixes #10952.

jdm committed:
Add a test for documents without browsing contexts loading stylesheets.

Ms2ger committed:
Merge pull request #3266 from w3c/sync_2595d093cde011b95a697b82c87f0d5324dd49b9

Merge pull request #3266 from sync_2595d093cde011b95a697b82c87f0d5324dd49b9

Ms2ger committed:
Merge pull request #3267 from w3c/sync_0277d157d63a7bed768d134f64f03b4594b44612

Merge pull request #3267 from sync_0277d157d63a7bed768d134f64f03b4594b44612

Ms2ger committed:
Merge pull request #3268 from w3c/sync_dda62d1be9ecc35d3f1ee028e8b4764580f3549d

Merge pull request #3268 from sync_dda62d1be9ecc35d3f1ee028e8b4764580f3549d

Ms2ger committed:
Merge pull request #3269 from w3c/sync_42bfe56f51fc7bc909cd847706aa3de1b1775216

Merge pull request #3269 from sync_42bfe56f51fc7bc909cd847706aa3de1b1775216

mustaqahmed committed:
Added tests for suppressed mouse events on canceled pointerdown (#3202)

Two separate tests: one for the general tap case and another for the mouse-drag case.

Doesn't attempt to test the behavior for touch drag because the spec and implementations often don't yet match (w3c/pointerevents/#7)

jdm committed:
Update tools submodule. (#3270); r=Ms2ger

engelke committed:
Merge pull request #6 from w3c/master

Catch up to W3C master

Commit:
Uses correct error field

jimsch committed:
Merge pull request #3272 from engelke/master

Fixed which error should be thrown.

Looks fine

halindrome committed:
This directory is not for the annotation-model project.

halindrome committed:
Merge branch 'master' of github.com:Spec-Ops/web-platform-tests

halindrome commented:
Git has defeated me whilst trying to squash these commits. I am killing this PR and will do a new one against a rebase'd tree.
